### DISCUSSION TITLE
Include alarm functionality in journals

### DISCUSSION THREAD

**Discussion by DickinsonSa** [TechbeeAT/jtxBoard#242] [Ideas]
# Proposed enhancement. 

**Overview** :

The capability to schedule alarms is currently present - and works well, within 'tasks'. There is however no such feature within 'journal'.

**Problem** :

To be notified of a journal / calendar event a task must be created. This pre-supposes that an upcoming event is noted and a corresponding task created. Too much work and open to forgetfulness.

Enhancement :

Implement the same 'alarm' functionality within the journal section.

**Comment by patrickunterwegs** [comment]
Hey @DickinsonSa , 
You're right, it would absolutely make sense to enable the alarms for journals (technically the element is just hidden for journals). The problem is the standard:

> The "VALARM" calendar component MUST only appear within either a "VEVENT" or "VTODO" calendar component. "VALARM" calendar components cannot be nested. Multiple mutually independent

So the tasks are VTODOs, the journals and notes are VJOURNALs. But it seems like the standard explicitly excludes VALARMs from VJOURNALs...

Some time ago this was even reported as a bug (I had overlooked this restriction). Even though it could easily be done, it could lead to errors on the servers if they have a strict implementation of the standard.
So I'm afraid that even I'm with you in this matter, it is better to stick to the standard to avoid any other errors when the export or sync features are used!

**Comment by DickinsonSa** [comment]
Hi Patrick, 

Thanks for the quick response. 

Just a thought: are subtasks VTODO components? Could they not have alarms? 

Just a thought to try to force an upcoming journal event to kick off some (any) kind of alarm / reminder. 

Can't think of any other way than improve my memory!

**Comment by patrickunterwegs** [comment]
Yes, subtasks are VTODOs, so basically it would be possible to add a subtask with an alarm. 
Or did you mean to have some more autimated functionality to do that? But this might get confusing if suddenly a subtask is created when actually they want an alarm...

**Comment by DickinsonSa** [comment]
No, to keep it simple I think: extend the functionality of the subtask to have an alarm property. 

This way tasks themselves have alarms, also subtasks (of those tasks) have their own alarms.

This should satisfy the standards (I hope) and also enables journal entries to have alarms (albeit by having a subtask with an alarm). 

So when creating a journal entry, simply create a subtask/s also (with an earlier date) as a reminder. 

Plus, when recurring tasks in journal entries is fixed it should be excellent. 

If this satisfies you and its doable, I can raise it as a seperate issue for consideration as an enhancement.

**Comment by patrickunterwegs** [comment]
Sounds good 🙂 

> Plus, when recurring tasks in journal entries is fixed it should be excellent.

You mean in general that subtasks should be enabled to be recurring? 
I can consider this in 2.3 where I'm anyway refactoring the recurring handling (technically - to better fulfill the standard and to  be better compatible with Thunderbird).

And in 2.4 I would like to technically improve alarms and also enable recurring alarms.

**Comment by DickinsonSa** [comment]
> You mean in general that subtasks should be enabled to be recurring? 

I was specifically thinking of bug #233 which is factored in for a later date. 

Many thanks. 😁

**Comment by patrickunterwegs** [comment]
Ah, alright, I unserstand 😉 
I also just discovered the bug, that basically it was possible to create recurring subtasks, but they disappeared, I just fixed this for 2.2.1 in #240 

> Plus, when recurring tasks in journal entries is fixed it should be excellent.

I am not so sure if this would solve your request. The problem is that subtasks are independent entries, the subtask does not inherit any date from the parent entry. So if you create a recurring journal with subtasks that have an alarm at a specific date, the date for ALL subtasks would be the same (unless I add a special logic for that, but I'm not sure if this is such a good idea...)

I'll also transform this issue into a discussion it's more about finding what should be done 😉

**Reply by DickinsonSa** [reply]
A discussion is a good way forward. It will save having to revisit logic further down the line.
